### DISCUSSION TITLE
Add use.pageController to Rearch

### DISCUSSION THREAD

**Discussion by Kypsis** [GregoryConrad/rearch-dart#110] [General]
The `SideEffectRegistrar` and `WidgetSideEffectRegistrar` are brilliant ways to approach extensibility. Since I had need for a `PageController` I used:

```dart
extension WidgetSideEffectRegistrarX on WidgetSideEffectRegistrar {
  SideEffectRegistrar get use => this;

  PageController pageController({
    int initialPage = 0,
    bool keepPage = true,
    double viewportFraction = 1.0,
    void Function(ScrollPosition)? onAttach,
    void Function(ScrollPosition)? onDetach,
  }) {
    final controller = use.memo(
      () => PageController(
        initialPage: initialPage,
        keepPage: keepPage,
        viewportFraction: viewportFraction,
        onAttach: onAttach,
        onDetach: onDetach,
      ),
    );
    use.effect(() => controller.dispose, [controller]);

    return controller;
  }
}
```

But would probably be handy to have it in Rearch itself.

**Comment by GregoryConrad** [comment]
@Kypsis A PR would be welcome! Feel free to add that to the bottom of packages/flutter_rearch/lib/src/side_effects.dart (along with a basic test or two)

**Reply by Kypsis** [reply]
Added PR [https://github.com/GregoryConrad/rearch-dart/pull/111](https://github.com/GregoryConrad/rearch-dart/pull/111)
